packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -412,12 +412,16 @@ export class DatabaseProxy extends cdk.Resource
     let dbInstanceIdentifiers: string[] | undefined;
     if (bindResult.dbInstances) {
       // support for only single instance
-      dbInstanceIdentifiers = [bindResult.dbInstances[0].instanceIdentifier];
+      const dbInstance = bindResult.dbInstances[0];
+      this.node.addDependency(dbInstance);
+      dbInstanceIdentifiers = [dbInstance.instanceIdentifier];
     }
 
     let dbClusterIdentifiers: string[] | undefined;
     if (bindResult.dbClusters) {
-      dbClusterIdentifiers = bindResult.dbClusters.map((c) => c.clusterIdentifier);
+      const dbClusters = bindResult.dbClusters;
+      dbClusters.forEach((cluster) => this.node.addDependency(cluster));
+      dbClusterIdentifiers = dbClusters.map((cluster) => cluster.clusterIdentifier);
```

Comment:
I think we're adding the dependency in the wrong place - I think they should be on the `ProxyTargetGroup`, not on the `Proxy` itself.

So, please revert these changes, and instead have something like this below:

```ts
    const proxyTargetGroup = new CfnDBProxyTargetGroup(this, 'ProxyTargetGroup', {
      // same as today...
    });
    bindResult.dbClusters?.forEach((c) => proxyTargetGroup.node.addDependency(c));
```